Firebase.h

```diff
@@ -25,49 +25,108 @@
 #include <WiFiClientSecure.h>
 #include <ESP8266HTTPClient.h>
 
-// FirebaseError represents a Firebase API error with a code and a
-// message.
-class FirebaseError {
+//TODO(edcoyne) split these into multiple files.
+
+// Result from call to Firebase backend. ALWAYS check isError() before
+// expecting any data.
+class FirebaseResult {
  public:
-  operator bool() const { return _code < 0; }
-  int code() const { return _code; }
-  const String& message() const { return _message; }
-  void reset() { set(0, ""); }
-  void set(int code, const String& message) {
-    _code = code;
-    _message = message;
+  FirebaseResult(int status);
+  FirebaseResult(int status, const String& response);
+  FirebaseResult(const FirebaseResult& result);
+
+  // True if there was an error completeing call.
+  bool isError() const;
```

Comment:
I liked the idea to have a different error object you should check like a bool with `fbase.error()`, even if we introduce a separate result maybe we could result that concept.

auto result = fbase.get();
if (result) {
  Serial.println(result.string());
}